### DISCUSSION TITLE
Multi Page form never renders page 2

### DISCUSSION THREAD

**Discussion by scott-davidjones** [solspace/craft-freeform#1528] [Questions & Answers] [answered]
Hi All, 

I have inherited a site which uses the following:
Craft: 4.12.3
FreeForm: 5.6.2
Blitz: 4.23.0

There is a single multipage form consisting of 2 pages. It uses a `tailwind.twig` template file and the form loads correctly.

The issue is with trying to get to page 2. On form submission i can see the HTML response that comes back consists of page 1 and not page 2.

The page is replaced correctly but with the wrong page HTML.

I have disabled Blitz and set `->enableTemplateCaching(false);` to no avail. clicking the submit button for a second time does cause errors to display based on the page 2 required inputs not being populated. 

I cannot see any reason for the page 1 HTML being returned insteead of page 2's HTML. 

Any thoughts? I have been trying to debug this for a few days :(

Update:
[template Code](https://pastebin.com/QaZ5QtkK)

**Comment by kjmartens** [comment]
Can you share what your template code looks like (where you load the form)? 🙂

**Reply by scott-davidjones** [reply]
sure, template code is [here](https://pastebin.com/QaZ5QtkK)

I created a completely new form, 2 pages, page 1 has a single input handle = 'name' and is required, page two has a single checkbox handle = 'ok'

and i get the same issue.

**Comment by scott-davidjones** [comment, marked as answer]
OK, i believe i have fixed it. 

After looking through the included freeform tailwind-3 template I noticed that you guys do `{{ rowMacro.render(form.rows, form) }}` which in turn does `{% for row in rows %}` inside [_row.twig](https://github.com/solspace/craft-freeform/blob/8711c53765d103f9a55fd83f34c8f6a0bfb60018/packages/plugin/src/templates/_templates/formatting/tailwind-3/_row.twig) 

I double checked the `tailwind.twig` file in the code base and the previous developer had used `{% for row in form %}` - swapping this out to `{% for row in form.rows %}` solved the problem and the form now correctly returns and renders page 2.

looking through the git logs it looks like this went from freeform 4 to freeform 5 not long ago and this functionality probably changed, not sure what the difference between `{% for row in form %}` and `{% for row in form.rows %}`  is but it appears to work..

**Reply by kjmartens** [reply]
Yes, there were several changes to how forms are rendered in their formatting templates. Glad you were able to figure this one out. 🙂
